// storage-function/lib/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.storageFunction = void 0;
var storage_1 = require("./storage");
var local_1 = require("./storage/local");
var session_1 = require("./storage/session");
exports.storageFunction = Object.freeze({
    toLocalStorage: local_1.toLocalStorage,
    fromLocalStorage: local_1.fromLocalStorage,
    getAllFromLocalStorage: local_1.getAllFromLocalStorage,
    getKeysFromLocalStorage: local_1.getKeysFromLocalStorage,
    getKeysCountFromLocalStorage: local_1.getKeysCountFromLocalStorage,
    removeFromLocalStorage: local_1.removeFromLocalStorage,
    clearLocalStorage: local_1.clearLocalStorage,
    toSessionStorage: session_1.toSessionStorage,
    fromSessionStorage: session_1.fromSessionStorage,
    getAllFromSessionStorage: session_1.getAllFromSessionStorage,
    getKeysFromSessionStorage: session_1.getKeysFromSessionStorage,
    getKeysCountFromSessionStorage: session_1.getKeysCountFromSessionStorage,
    removeFromSessionStorage: session_1.removeFromSessionStorage,
    clearSessionStorage: session_1.clearSessionStorage,
    clearStorage: storage_1.clearStorage,
    getAll: storage_1.getAll,
    getKeys: storage_1.getKeys,
    getKeysCount: storage_1.getKeysCount,
});
